src/components/functional/XButton.tsx

```diff
@@ -0,0 +1,39 @@
+import { FunctionalComponent, h } from "@stencil/core";
+import { JSXAttributes, JSXBase } from "@stencil/core/internal";
+import { Scale } from "../interfaces";
+
+export interface XButtonOptions extends JSXAttributes {
```

Comment:
maybe it should be `ClearButton`/`ResetButton` since the `x` icon is used for close as well?

![image](https://github.com/Esri/calcite-components/assets/1231455/625781b3-671f-4714-8723-e83fad2d59c3)

Reply:
Bounced names w/ @ashetland and we landed on x-button as it could be used for both clearing and closing. Is there another name that could represent clear/reset/close? Luckily, it's internal for now, so we can rename if we land on a better name.

Reply:
sounds good!